wordpress-mobile/WordPress-iOS
# Stats: Localize percentage strings

Fixes #18536

Percentage sign placement depends on a locale. At the moment, we include it localized string with an explanation but the string we provide for translators is hard to understand and work with and the placement of the percentage sign is not changed based on a locale.

### Solution

1. Remove percentage sign from localized strings
2. Use `NumberFormatter` with application's localization locale to build a number with percentage

- [x] Insights -> Most Popular Time card
- [x] Insights -> Views and Visitors card
- [x] Insights -> Total Likes card
- [x] Insights -> Total Comments card
- [x] Days/Weeks/Months/Years -> Overview Chart

### To test:

Note: At the moment new localized strings are not yet localized, so they will appear in English while percentage sign will be placed based on a current locale

1. Change your device language to Turkish from OS settings > General > Language & Region > iPad Language > Türkçe(Turkish)
3. Open Jetpack app.
4. Go to stats either using quick links or menu.
5. Switch to DAYS ("GÜNLER" in Turkish) tab.
6. Check the percent sign and notice it's the end of the number instead of the start.

## Regression Notes
1. Potential unintended areas of impact

None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

None

3. What automated tests I added (or what prevented me from doing so)
None

PR submission checklist:


Testing checklist:
